lightwood/encoders/categorical/onehot.py

```diff
@@ -60,9 +63,22 @@ def encode(self, column_data):
     def decode(self, encoded_data):
         encoded_data_list = encoded_data.tolist()
         ret = []
+        probs = []
 
         for vector in encoded_data_list:
             ohe_index = np.argmax(vector)
-
             ret.append(self._lang.index2word[ohe_index])
-        return ret
+
+            if self.predict_proba:
+                del(vector[UNCOMMON_TOKEN])
+                probs.append(softmax(vector).tolist())
+
+        if self.predict_proba:
+            # UNK not included in class_map nor belief distribution
+            if UNCOMMON_TOKEN != 0:
+                raise Exception("Uncommon token should be the first assigned token in the vocabulary, aborting.")
+            class_map = deepcopy(self._lang.index2word)
+            del(class_map[UNCOMMON_TOKEN])
+            return ret, probs, {k-1: v for k, v in class_map.items()}
```

Comment:
`{k-1: v for k, v in class_map.items() if k != UNCOMMON_TOKEN}`

should solve the need for a potentially expensive deepcopy (since done very often) and make the code less confuing (since that `del` is no longer needed)